tests/Unit/Core/Search/Builder/PersistFiltersBuilderTest.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\Search\Builder;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Employee\ContextEmployeeProviderInterface;
+use PrestaShop\PrestaShop\Core\Search\Builder\PersistFiltersBuilder;
+use PrestaShop\PrestaShop\Core\Search\Filters;
+use PrestaShopBundle\Entity\Repository\AdminFilterRepository;
+use Symfony\Component\HttpFoundation\Request;
+
+class PersistFiltersBuilderTest extends TestCase
+{
+    private const EMPLOYEE_ID = 42;
+    private const SHOP_ID = 51;
+
+    public function testBuildWithoutParameters(): void
+    {
+        $builder = new PersistFiltersBuilder(
+            $this->buildUnusedRepository(),
+            $this->buildUnusedEmployeeProviderMock(),
+            static::SHOP_ID
```

Comment:
```suggestion
            self::SHOP_ID
```
It's private, so can't be called outside its own scope :trollface: